azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -748,7 +767,7 @@ func resourceArmKubernetesClusterUpdate(d *schema.ResourceData, meta interface{}
 	if d.HasChange("api_server_authorized_ip_ranges") {
 		updateCluster = true
 		apiServerAuthorizedIPRangesRaw := d.Get("api_server_authorized_ip_ranges").(*schema.Set).List()
-		existing.APIServerAuthorizedIPRanges = utils.ExpandStringSlice(apiServerAuthorizedIPRangesRaw)
+		existing.ManagedClusterProperties.APIServerAccessProfile.AuthorizedIPRanges = utils.ExpandStringSlice(apiServerAuthorizedIPRangesRaw)
```

Comment:
there's the potential for `existing.ManagedClusterProperties.APIServerAccessProfile` to be nil depending on the API version being used here; instead it'd be better to assign an object to `APIServerAccessProfile` here:

```suggestion
		existing.ManagedClusterProperties.APIServerAccessProfile = &containers.APIServerAccessProfile{
			AuthorizedIPRanges: utils.ExpandStringSlice(apiServerAuthorizedIPRangesRaw),
		}
```

Reply:
Resolved in new commit